make push-agones-sdk-manifest fails

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via 
-->


**What happened**:

```
make push-agones-sdk-manifest
make[1]: Entering directory '/home/mark/workspace/agones/build'
docker manifest create --amend repository.dev/images/agones-sdk:1.49.0-dev-bdacd1d repository.dev/images/agones-sdk:1.49.0-dev-bdacd1d-linux-amd64  repository.dev/images/agones-sdk:1.49.0-dev-bdacd1d-windows_amd64-ltsc2019  repository.dev/images/agones-sdk:1.49.0-dev-bdacd1d-windows_amd64-ltsc2022 repository.dev/images/agones-sdk:1.49.0-dev-bdacd1d-linux-arm64
Created manifest list repository.dev/images/agones-sdk:1.49.0-dev-bdacd1d
for osversion in ltsc2019 ltsc2022; do \
        full_version=`docker manifest inspect mcr.microsoft.com/windows/servercore:$osversion | grep "os.version" | head -n 1 | awk -F\" '{print $4}'` || true; \
        docker manifest annotate --os windows --arch amd64 --os-version $full_version \
        repository.dev/images/agones-sdk:1.49.0-dev-bdacd1d repository.dev/images/agones-sdk:1.49.0-dev-bdacd1d-windows_amd64-$osversion; \
done
docker manifest push --purge repository.dev/images/agones-sdk:1.49.0-dev-bdacd1d
failed to put manifest repository.dev/images/agones-sdk:1.49.0-dev-bdacd1d: manifest invalid: Manifest references unknown manifest(s): agones-sdk/manifests/sha256:4745e59050276392776a898ae47e78257f045496211ebf9b64a0b2afb108756c
make[1]: *** [Makefile:719: push-agones-sdk-manifest] Error 1
make[1]: Leaving directory '/home/mark/workspace/agones/build'
make: *** [Makefile:714: push-agones-sdk-image] Error 2
```

This is my version:
```
❯ docker --version
Docker version 28.0.4, build b8034c0
```

**What you expected to happen**:

To work as it previously had. Previously I could also use `WITH_ARM=0` and `WITH_WINDOWS=0`, this doesn't seem to be working anymore.

**How to reproduce it (as minimally and precisely as possible)**:

`make -j 4 build-images && make push`

**Anything else we need to know?**:

**Environment**:
- Agones version:  dev
- Kubernetes version (use `kubectl version`): 1.31
- Cloud provider or hardware configuration: N/A
- Install method (yaml/helm): helm
- Troubleshooting guide log(s): N/A
- Others:

--- a/build/Makefile
+++ b/build/Makefile
@@ -711,16 +711,21 @@ endif
 ifeq ($(WITH_ARM64), 1)
 push-agones-sdk-image: push-agones-sdk-linux-image-arm64
 endif
-	$(MAKE) push-agones-sdk-manifest	
+	$(MAKE) push-agones-sdk-manifest
 
-push-agones-sdk-manifest:
+push-agones-sdk-manifest: 
 	docker manifest create --amend $(sidecar_tag) $(push_sidecar_manifest)
+	$(MAKE) push_windows_manifest
+	docker manifest push --purge $(sidecar_tag)
+
+push_windows_manifest:
+ifeq ($(WITH_WINDOWS), 1)
 	for osversion in $(WINDOWS_VERSIONS); do \
 		full_version=`docker manifest inspect ${BASE.windows}:$$osversion | grep "os.version" | head -n 1 | awk -F\" '{print $$4}'` || true; \
 		docker manifest annotate --os windows --arch amd64 --os-version $$full_version \
 		$(sidecar_tag) $(sidecar_tag)-windows_amd64-$$osversion; \
 	done
-	docker manifest push --purge $(sidecar_tag)
+endif
 
 push-agones-sdk-windows: WINDOWS_DOCKER_PUSH_ARGS=--push
 push-agones-sdk-windows: build-agones-sdk-image-windows

